### PR TITLE
Cleanup TiDAGRequest code

### PR DESCRIPTION
**Pull request by birdstorm**
There were some confusing code in `TiDAGRequest `logic. This PR cleans up some unused code and make logic clearer.

### CODE REVIEW

**Comment by birdstorm** [comment]
@zhexuany PTAL

File: tikv-client/src/main/java/com/pingcap/tikv/meta/TiDAGRequest.java [hunk 1]
```diff
@@ -357,34 +357,25 @@ else if (col.getColumnInfo().isPrimaryKey() && tableInfo.isPkHandle()) {
       tblScanBuilder.setTableId(id);
       // Step1. Add columns to first executor
       int realOffset = 0;
-      for (int i = 0; i < getFields().size(); i++) {
-        ColumnRef col = getFields().get(i);
+      for (ColumnRef col : getFields()) {
+        int offset = realOffset;
+        realOffset++;
```
Comment (zhexuany) [inline tikv-client/src/main/java/com/pingcap/tikv/meta/TiDAGRequest.java]: - - What's wrong with the original fix?

Comment (birdstorm) [reply tikv-client/src/main/java/com/pingcap/tikv/meta/TiDAGRequest.java]: original code wasn't coooooool!

**Pull request merged**